pkg/archive/filter.go

```diff
@@ -26,6 +26,19 @@ func getFilterPath(name string) string {
 	return path.(string)
 }
 
+type errorRecordingReader struct {
+	r   io.Reader
+	err error
+}
+
+func (r *errorRecordingReader) Read(p []byte) (int, error) {
+	n, err := r.r.Read(p)
+	if err != nil && err != io.EOF {
+		r.err = err
```

Comment:
I think this should prefer the first error, not the last one.

Reply:
yes that is better, changed now

Reply:
… but still not store `err == nil`?

Reply:
… never mind, this works without an explicit check for `err != nil` correctly.